### PR TITLE
Stream transport should enable verify_peer_name when possible

### PR DESCRIPTION
**Pull request by wilsonge**
### Summary of Changes
Enables `verify_peer_name` for the stream driver (which ensures the ssl certificate matches the domain name being queried). This does default to true since it was introduced in PHP 5.6 - but let's be explicit!

### Testing Required
Ensure the stream http driver continues to work as expected (with the additional "security" mechanism enabled)

### Documentation Changes Required
None

### CODE REVIEW

**Review by mbabker — APPROVED** [review]

File: libraries/joomla/http/transport/stream.php [hunk 1]
```diff
@@ -166,6 +166,12 @@ public function request($method, JUri $uri, $data = null, array $headers = null,
 			)
 		);
 
+		// Ensure the ssl peer name is verified where possible
+		if (version_compare(PHP_VERSION, '5.6.0') >= 0)
+		{
+			$streamOptions['ssl']['verify_peer_name'] = true;
```
Comment (roland-d) [inline libraries/joomla/http/transport/stream.php]: @wilsonge It looks like ```$streamOptions``` isn't used anywhere. Are you sure the variable name is correct?

Comment (wilsonge) [reply libraries/joomla/http/transport/stream.php]: Whoops. Apparently this wasn't as simple as in the framework! Fixed now. Thanks for the review!

**Comment by roland-d** [comment]
@wilsonge Should we do the same for cURL or not?

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on ef2d601188e92aa66776f2b0f84806e0a3a53cbd<br /><br />Before the patch the SSL verification isn't explicitly set. After the patch the SSL verification is explicitly set and the request still works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16501">issues.joomla.org/tracker/joomla-cms/16501</a>.</sub>

**Comment by brianteeman** [comment]
@wilsonge can you fix the conflicts please

**Comment by brianteeman** [comment]
@wilsonge reminder to fix conflicts

**Comment by wilsonge** [comment]
Thanks for the prod! All done :)

**Comment by HLeithner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16501">issues.joomla.org/tracker/joomla-cms/16501</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**